changebot/github/github_api.py

```diff
@@ -99,6 +101,32 @@ def get_file_contents(self, path_to_file):
         contents_base64 = response.json()['content']
         return base64.b64decode(contents_base64).decode()
 
+    # TODO: Need caching per repo so don't have to load this all the time.
+    def get_user_config(self, path_to_file='astropybotrules.yaml'):
+        """
+        Load user configuration for bot.
+
+        Parameters
+        ----------
+        path_to_file : str
+            Configuration file in YAML format in the repository.
+            If not given or invalid, default is used.
+
+        Returns
+        -------
+        cfg : dict
+            Configuration parameters.
+
+        """
+        try:
+            file_content = self.get_file_contents(path_to_file)
```

Comment:
Brain storm: Maybe before loading from file here, we can check the cache first (whatever that may be, is there an existing property I can piggyback on?). I briefly thought about maybe changing API to accept only desired config keyword and only to return its value instead of a `dict` but decided against it, as that way is inflexible in case user wants to extract multiple config for some bot functionality; The config won't get so big such that returning the whole `dict` would be a problem, right?

Reply:
I think the whole dict is fine, indeed it's not going to grow much. For the cache, I would just make a global variable in ``github_api`` that is a dictionary where each key is a repository name. Something similar is done in ``github_auth`` to keep tokens for each repo (but we shouldn't use the same variable)